zig not linking macos frameworks

### Zig Version

0.12.0-dev.596+2adb932ad

### Steps to Reproduce and Observed Behavior

- macos14, aarch/m1 cpu

---

```c
/* main.c */
#include <SDL2/SDL.h>

int main() {
    SDL_Init(SDL_INIT_VIDEO);
    SDL_Quit();
    printf("success\n");
    return 0;
}
```
- compiles fine with `cc`
```shell
cc -o main main.c -I/opt/homebrew/include -L/opt/homebrew/lib \
  /opt/homebrew/lib/libSDL2.a \
  -l z -l bz2 -l iconv \
  -framework Foundation -framework Cocoa -framework CoreGraphics -framework IOKit \
  -framework Metal -framework Carbon -framework CoreServices \
  -framework GameController -framework CoreHaptics -framework ForceFeedback \
  -framework CoreVideo -framework AudioToolbox -framework CoreAudio
./main
```
- but, use `zig cc` instead and it fails to find symbols from macos frameworks
```shell
zig cc -o main main.c -I/opt/homebrew/include -L/opt/homebrew/lib \
  /opt/homebrew/lib/libSDL2.a \
  -l z -l bz2 -l iconv \
  -framework Foundation -framework Cocoa -framework CoreGraphics -framework IOKit \
  -framework Metal -framework Carbon -framework CoreServices \
  -framework GameController -framework CoreHaptics -framework ForceFeedback \
  -framework CoreVideo -framework AudioToolbox -framework CoreAudio
./main
```

- similarly, converting the cmd to a `build.zig`, and the `main.c` to a `main.zig` fails in the same way

---

- **repo with all the minimal code examples here:** [🔗](https://github.com/nogira/minimum-viable-zig-static-linking-bug)

---

some of the countless missing symbol errors:
```shell
error: undefined reference to symbol _objc_msgSend$initWithContentRect:styleMask:backing:defer:screen:
    note: referenced in /opt/homebrew/Cellar/sdl2/2.28.3/lib/libSDL2.a(SDL_cocoawindow.o)
error: undefined reference to symbol _objc_msgSend$windowDidExitFullScreen:
    note: referenced in /opt/homebrew/Cellar/sdl2/2.28.3/lib/libSDL2.a(SDL_cocoawindow.o)
error: undefined reference to symbol _objc_msgSend$addItem:
    note: referenced in /opt/homebrew/Cellar/sdl2/2.28.3/lib/libSDL2.a(SDL_cocoaevents.o)
error: undefined reference to symbol _objc_msgSend$activateIgnoringOtherApps:
    note: referenced in /opt/homebrew/Cellar/sdl2/2.28.3/lib/libSDL2.a(SDL_cocoawindow.o)
error: undefined reference to symbol _objc_msgSend$finishLaunching
    note: referenced in /opt/homebrew/Cellar/sdl2/2.28.3/lib/libSDL2.a(SDL_cocoaevents.o)
```

### Expected Behavior

expected successful compilation

--- a/src/link/MachO.zig
+++ b/src/link/MachO.zig
@@ -707,6 +707,17 @@ fn accessLibPath(
         return true;
     }
 
+    noextension: {
+        test_path.clearRetainingCapacity();
+        try test_path.writer().print("{s}" ++ sep ++ "{s}", .{ search_dir, lib_name });
+        try checked_paths.append(try gpa.dupe(u8, test_path.items));
+        fs.cwd().access(test_path.items, .{}) catch |err| switch (err) {
+            error.FileNotFound => break :noextension,
+            else => |e| return e,
+        };
+        return true;
+    }
+
     return false;
 }
 

--- a/src/main.zig
+++ b/src/main.zig
@@ -6929,9 +6929,9 @@ fn accessFrameworkPath(
 ) !bool {
     const sep = fs.path.sep_str;
 
-    for (&[_][]const u8{ "tbd", "dylib" }) |ext| {
+    for (&[_][]const u8{ ".tbd", ".dylib", "" }) |ext| {
         test_path.clearRetainingCapacity();
-        try test_path.writer().print("{s}" ++ sep ++ "{s}.framework" ++ sep ++ "{s}.{s}", .{
+        try test_path.writer().print("{s}" ++ sep ++ "{s}.framework" ++ sep ++ "{s}{s}", .{
             framework_dir_path,
             framework_name,
             framework_name,

